core-bundle/src/Security/Authentication/AuthenticationEntryPoint.php

```diff
@@ -50,13 +53,31 @@ public function start(Request $request, ?AuthenticationException $authException
 
     private function redirectToBackend(Request $request): Response
     {
-        $url = $this->router->generate(
+        // No redirect parameter required if the 'contao_backend' route was requested
+        // without any parameters.
+        if ('contao_backend' === $request->attributes->get('_route') && [] === $request->query->all()) {
+            $loginParams = [];
+        } else {
+            $loginParams = ['redirect' => $request->getUri()];
+        }
+
+        $location = $this->router->generate(
             'contao_backend_login',
-            ['redirect' => $request->getUri()],
+            $loginParams,
             UrlGeneratorInterface::ABSOLUTE_URL
         );
 
-        $location = $this->uriSigner->sign($url);
+        // No URL signing required if we do not have any parameters
+        if ([] !== $loginParams) {
+            $location = $this->uriSigner->sign($location);
+        }
+
+        // Our back end login controller will redirect based on the 'redirect' parameter,
+        // ignoring Symfony's target path session value. Thus we remove the session variable
+        // here in order to not send an unnecessary session cookie.
+        if ($request->hasSession()) {
+            $this->removeTargetPath($request->getSession(), 'contao_backend');
```

Comment:
could you point to the place where this session value is added? I'd like to understand why would that be the case on this page but not "everywhere"?

Reply:
It is set [here](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Http/Firewall/ExceptionListener.php#L206-L208) when accessing the `contao_backend` route without being logged in.